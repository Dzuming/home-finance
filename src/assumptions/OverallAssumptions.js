import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { makeGetOverallAssumption } from '../helpers/selectors';
import * as assumptionActions from '../actions/assumptionActions';
import { connect } from 'react-redux';
import { Card, CardContent, Grid, Typography } from 'material-ui';

class OverallAssumptions extends Component {
  componentDidMount() {
    this.props.actions.fetchOverallAssumptions();
  }

  render() {
    const { overallAssumptions } = this.props;
    return (
      <React.Fragment>
        <Grid container spacing={0}>
          <Grid item xs={6}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  Assumption
                </Typography>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={6}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  Value
                </Typography>
              </CardContent>
            </Card>
          </Grid>
        </Grid>
        {overallAssumptions.map((assumption, index) => (
          <Grid key={index} container spacing={0}>
            <Grid item xs={6}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.name}
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={6}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.value} z≈Ç
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
          </Grid>
        ))}
      </React.Fragment>
    );
  }
}

OverallAssumptions.propTypes = {
  actions: PropTypes.shape({
    fetchOverallAssumptions: PropTypes.Func,
  }),
  overallAssumptions: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  overallAssumptions: makeGetOverallAssumption(state),
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(assumptionActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(OverallAssumptions);
