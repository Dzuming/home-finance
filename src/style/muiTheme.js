import { createMuiTheme } from 'material-ui/styles';

export const theme = createMuiTheme({
  status: {
    success: '#4CAF50',
    warning: '#F44336',
  },
});
