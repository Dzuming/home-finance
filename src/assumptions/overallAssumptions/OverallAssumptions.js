import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { makeGetOverallAssumption } from '../../selectors/assumptions';
import * as assumptionActions from '../../actions/assumptionActions';
import { connect } from 'react-redux';
import { Card, CardContent, Grid, Typography } from 'material-ui';
import CardList from '../../components/commons/CardList';

class OverallAssumptions extends Component {
  componentDidMount() {
    this.props.actions.fetchOverallAssumptions();
  }

  render() {
    const { overallAssumptions } = this.props;
    return (
      <React.Fragment>
        <Grid container spacing={0}>
          <CardList name={'Assumption'} gridSize={6} />
          <CardList name={'Value'} gridSize={6} />
        </Grid>
        {overallAssumptions.map((assumption, index) => (
          <Grid key={index} container spacing={0}>
            <CardList name={assumption.name} gridSize={6} />
            <CardList name={`${assumption.value} zÅ‚`} gridSize={6} />
          </Grid>
        ))}
      </React.Fragment>
    );
  }
}

OverallAssumptions.propTypes = {
  actions: PropTypes.shape({
    fetchOverallAssumptions: PropTypes.Func
  }),
  overallAssumptions: PropTypes.array.isRequired
};

const mapStateToProps = state => ({
  overallAssumptions: makeGetOverallAssumption(state)
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(assumptionActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(OverallAssumptions);
