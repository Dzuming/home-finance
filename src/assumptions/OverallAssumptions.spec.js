import React from 'react';
import { create } from 'react-test-renderer';
import OverallAssumptions from './OverallAssumptions';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { theme } from '../style/muiTheme';
import { MuiThemeProvider } from 'material-ui';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({
  assumptions: {
    overall: [],
  },
  user: [],
  login: { isAuthenticated: true },
});
const component = create(
  <MuiThemeProvider theme={theme}>
    <Provider store={store}>
      <OverallAssumptions />
    </Provider>
  </MuiThemeProvider>,
);

describe('<MonthlyAssumption  />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
