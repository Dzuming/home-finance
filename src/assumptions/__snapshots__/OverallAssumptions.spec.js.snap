// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthlyAssumption  /> should match snapshot 1`] = `
<div
  className="MuiGrid-typeContainer-1"
>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-6-32"
  >
    <div
      className="MuiPaper-root-92 CardList-card-91 MuiPaper-shadow2-96 MuiPaper-rounded-93"
    >
      <div
        className="MuiCardContent-root-119"
      >
        <aside
          className="MuiTypography-root-120 MuiTypography-body2-128 MuiTypography-colorInherit-139 MuiTypography-alignCenter-133"
        >
          Assumption
        </aside>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-6-32"
  >
    <div
      className="MuiPaper-root-92 CardList-card-91 MuiPaper-shadow2-96 MuiPaper-rounded-93"
    >
      <div
        className="MuiCardContent-root-119"
      >
        <aside
          className="MuiTypography-root-120 MuiTypography-body2-128 MuiTypography-colorInherit-139 MuiTypography-alignCenter-133"
        >
          Value
        </aside>
      </div>
    </div>
  </div>
</div>
`;
